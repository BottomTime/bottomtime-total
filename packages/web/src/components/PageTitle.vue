<template>
  <section class="hero is-dark">
    <div class="hero-body">
      <div class="container">
        <p class="title">{{ title }}</p>
        <p v-if="subtitle" class="subtitle">{{ subtitle }}</p>
        <slot></slot>
      </div>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { onMounted } from 'vue';

interface PageTitleProps {
  title: string;
  subtitle?: string;
}

const props = defineProps<PageTitleProps>();

onMounted(() => {
  document.title = `BottomTime | ${props.title}`;
});
</script>
