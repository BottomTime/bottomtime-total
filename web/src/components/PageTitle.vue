<template>
  <section class="hero is-info">
    <div class="hero-body">
      <div class="container">
        <p class="title">{{ title }}</p>
        <p v-if="subtitle" class="subtitle">{{ subtitle }}</p>
      </div>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { defineProps, onMounted } from 'vue';

interface PageTitleProps {
  title: string;
  subtitle?: string;
}

const props = defineProps<PageTitleProps>();

onMounted(() => {
  document.title = `BottomTime | ${props.title}`;
});
</script>
