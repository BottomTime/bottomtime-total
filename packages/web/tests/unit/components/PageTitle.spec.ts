import { shallowMount } from '@vue/test-utils';

import PageTitle from '@/components/PageTitle.vue';

describe('PageTitle Component', () => {
  it('Will display a title and subtitle', () => {
    const title = 'Title';
    const subtitle = 'A Witty Subtitle';

    const wrapper = shallowMount(PageTitle, {
      props: { title, subtitle },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(document.title).toEqual(`BottomTime | ${title}`);
  });

  it('Will display without a subtitle', () => {
    const title = 'Title';

    const wrapper = shallowMount(PageTitle, {
      props: { title },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
