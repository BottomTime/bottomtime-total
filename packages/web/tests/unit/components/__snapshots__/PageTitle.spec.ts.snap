// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Component Will display a title and subtitle 1`] = `
<section class="hero is-dark">
  <div class="hero-body">
    <div class="container">
      <p class="title is-capitalized">Title</p>
      <p class="subtitle">A Witty Subtitle</p>
    </div>
  </div>
</section>
`;

exports[`PageTitle Component Will display without a subtitle 1`] = `
<section class="hero is-dark">
  <div class="hero-body">
    <div class="container">
      <p class="title is-capitalized">Title</p>
      <!--v-if-->
    </div>
  </div>
</section>
`;
